[Maps] search bar docs (#34315)
elastic/kibana: docs/maps/search.asciidoc

[role="xpack"]
[[maps-search]]
== Searching Your Data

The **Maps** application embeds the query bar with autocomplete for real-time ad hoc search.
All layers with {es} sources are updated when you submit a search request.

image::maps/images/global_search_bar.png[]

[float]

> === Layer with {es} source

Reviewer: This feels like it belongs in a separate doc with descriptions of when to use each.  I'm also not sure how the content about layers here relates to the sections about the Heat map layer, Tile layer, and Vector layer.

If you want to include information about layers here, provide more context, especially in relation to search. Are these the only types of layers you can search?

Author: > If you want to include information about layers here, provide more context, especially in relation to search. Are these the only types of layers you can search?

Yes, this section is defining which layers will be effected by the global search bar

Reviewer: Maybe you can add some wording around that.

You can create a layer with {es} source from the following:

- <<vector-layer>> with *Documents* source.

- <<vector-layer>> with *Grid aggregation* source.

- <<heatmap-layer>> with *Grid aggregation* source.

- <<vector-layer>> with <<terms-join>>.

[float]

> === Searching across multiple indexes

Reviewer: ```suggestion
=== Searching across multiple indices
```

Author: I used `indexes` to be consistent with other places in Elasticsearch docs, such as https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-index-field.html. Is there an effort to migrate `indexes` to `indices`? Which is preferred?

Reviewer: The Elasticsearch writing style guide says to use "indices."

> Your map may contain multiple {es} indexes.

Reviewer: ```suggestion
Your map might contain multiple {es} indices.
```

> This can occur when your map contains two or more layers with {es} sources from different indexes.

Reviewer: ```suggestion
This might occur when your map contains two or more layers with {es} sources from different indices.
```

> This can also occur with a single layer with an {es} source and a <<terms-join>>.

Reviewer: ```suggestion
This might also occur with a single layer with an {es} source and a <<terms-join>>.
```

> 

Reviewer: It would be nice to show an example search here, before talking about empty layers.

> Searching across multiple indexes can sometimes result in unanticipated empty layers.

Reviewer: ```suggestion
Searching across multiple indices can sometimes result in empty layers.
```

> The most common cause for empty layers are searches for a field that exist in one index but does not exist in other indexes.

Reviewer: ```suggestion
The most common cause for empty layers are searches for a field that exists in one index but does not exist in other indices.
```

> Add {ref}/mapping-index-field.html[_index] to your search to include documents from indexes that do not contain a search field.

Reviewer: ```suggestion
Add {ref}/mapping-index-field.html[_index] to your search to include documents from indices that do not contain a search field.
```

[role="xpack"]
[[maps-search]]

> == Searching Your Data

Reviewer: ```suggestion
== Searching your data
```

The **Maps** application embeds the query bar for real-time ad hoc search.
Only layers requesting data from {es} are filtered when you submit a search request.

You can create a layer that requests data from {es} from the following:

* <<vector-layer>> with:

> ** Documents source.

Reviewer: ```suggestion
** Documents source
```

> ** Grid aggregation source.

Reviewer: ```suggestion
** Grid aggregation source
```

> ** <<terms-join>>.

Reviewer: ```suggestion
** <<terms-join>>
```

> * <<heatmap-layer>> with Grid aggregation source.

Reviewer: ```suggestion
* <<heatmap-layer>> with Grid aggregation source
```

[role="screenshot"]
image::maps/images/global_search_bar.png[]

[float]
=== Searching across multiple indices

Your map might contain multiple {es} indices.
This can occur when your map contains two or more layers with {es} sources from different indices.
This can also occur with a single layer with an {es} source and a <<terms-join>>.

Searching across multiple indices might sometimes result in empty layers.
The most common cause for empty layers are searches for a field that exists in one index, but does not exist in other indices.
Add {ref}/mapping-index-field.html[_index] to your search to include documents from indices that do not contain a search field.

For example, suppose you have a vector layer showing the `kibana_sample_data_logs` documents
and another vector layer with `kibana_sample_data_flights` documents.
(See <<add-sample-data, adding sample data>>

> to install `kibana_sample_data_logs` and `kibana_sample_data_flights` indices.)

Reviewer: ```suggestion
to install the `kibana_sample_data_logs` and `kibana_sample_data_flights` indices.)
```

If you query for
--------------------------------------------------
machine.os.keyword : "osx"
--------------------------------------------------
the `kibana_sample_data_flights` layer is empty because the index
`kibana_sample_data_flights` does not contain the field `machine.os.keyword` and no documents match the query.

[role="screenshot"]
image::maps/images/global_search_multiple_indices_query1.png[]

If you instead query for
--------------------------------------------------
machine.os.keyword : "osx" or _index : "kibana_sample_data_flights"
--------------------------------------------------

> the `kibana_sample_data_flights` layer is not empty.

Reviewer: ```suggestion
the `kibana_sample_data_flights` layer includes data.
```
